components/tls/Makefile

```diff
@@ -18,7 +18,7 @@ lint: ${.GOLANGCILINT_PATH}/golangci-lint
 
 .PHONY: build
 build: test
-	go build ./pkg/...
+	go build -trimpath -ldflags="-s -w" ./pkg/...
```

Comment:
Using -w (strip DWARF info) is justified, but I would argue against -s (stripping debug symbols). The debug symbols can offer us good information if things fail in production (i.e the customer gets a stack trace) and we need to figure what went wrong from looking at the logs. I know `-w -s` is common for production golang binaries, but I don't see the value of -s compared to the disadvantages.

Furthermore, the debug symbols alone do not take much space, and shouldn't really reveal sensitive information either.

Ideally we can also create detached debuginfo files (i.e based on no -w) per release for ourselves to go along the binaries in case we ever need to debug challenging issues (let's discuss this outside this PR)

Reply:
I've updated the PR to remove the -s flag, as per the requirement. My initial thought was that since most things run in containers, we might not often need to check the assembly output. Even in the case of a code panic, we would still get the stack trace.

Reply:
It wasn't exactly a requirement, more of a good thing to have. It's not about looking at the assembly output of the binaries, but about stack traces that may be printed as part of errors or during panics.

Not having the symbols would of course not stop the stack traces from appearing, but it will make them less useful: instead of them showing the function names & location across all the stack frames, we would just get memory addresses and byte offsets.